stacbuilder/builder.py

```diff
@@ -496,13 +507,20 @@ def save_collection(self) -> None:
     def providers(self):
         return [p.to_provider() for p in self._collection_config.providers]
 
-    def _create_empty_collection(self) -> None:
+    def create_empty_collection(self, group: Optional[str | int] = None) -> None:
         """Creates a STAC Collection with no STAC items."""
-
         coll_config: CollectionConfig = self._collection_config
+
+        if group:
+            id = coll_config.collection_id + f"_{group}"
+            title = coll_config.title + f" {group}"
+        else:
+            id = coll_config.collection_id
+            title = coll_config.title
+
```

Comment:
I think that in this case the STACCollectionBuilder needs to know the group to add to suffix to the title and id (so that the id's are unique) for the creation of the collection.
In the current implementation the id is derived from the config, so otherwise we would need to create a new config for every group collection. Passing an optional arguments seems like a less confusing solution.

Reply:
Ah I see. 

I was thinking more along the lines of not bothering to change the coll_config, and passing ID and title directly to create_empty_collection, regardless of what the ID and title are in the config. We can still use the rest of the config values. 

The CollectionConfig is just something we read from file and copy data from. We probably don't need to update it for each child collection.
If really necessary we can just change it in memory without saving anything to file for the child collections. I would definitely try to avoid that kind of extra complexity. After all, CollectionConfig is basically like a dataclass, but with some extra stuff to serialize it.

But that said, this doesn't have to stall this PR. 
This is good enough, especially if dealing with issues on the CollectionConfig makes it more complicated.
If necessary, that could be addressed later.